### PR TITLE
SONAR-6226 Apply various feedback

### PR DESCRIPTION
**Pull request by julienlancelot**

### CODE REVIEW

**Comment by julienlancelot** [comment]
Associated PR on GitHub plugin : https://github.com/SonarSource/sonar-auth-github/pull/3

File: server/sonar-web/src/main/webapp/WEB-INF/app/views/sessions/_form.html.erb [hunk 1]
```diff
@@ -7,9 +7,10 @@
       <% auth_providers.each do |provider| %>
         <li>
           <a href="<%= ApplicationController.root_context -%>/sessions/init/<%= provider.getKey().to_s %>"
```
Comment (simonbrandhof) [inline server/sonar-web/src/main/webapp/WEB-INF/app/views/sessions/_form.html.erb]: No need for <%= ApplicationController.root_context -%> anymore (same for line 13)

File: sonar-plugin-api/src/main/java/org/sonar/api/server/authentication/Display.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  }
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  public static class Builder {
+
+    private String iconPath;
+    private String backgroundColor;
```
Comment (simonbrandhof) [inline sonar-plugin-api/src/main/java/org/sonar/api/server/authentication/Display.java]: It's not always appropriate to define a color. I suggest to use a default color that can be overridden.

File: sonar-plugin-api/src/main/java/org/sonar/api/server/authentication/Display.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+  private final String backgroundColor;
+
+  private Display(Builder builder) {
+    this.iconPath = builder.iconPath;
+    this.backgroundColor = builder.backgroundColor;
+  }
+
+  /**
+   * URL path to the provider icon, as deployed at runtime, for example "/static/authgithub/github.svg" (in this
+   * case "authgithub" is the plugin key. Source file is "src/main/resources/static/github.svg"). Must not be blank.
```
Comment (simonbrandhof) [inline sonar-plugin-api/src/main/java/org/sonar/api/server/authentication/Display.java]: "Must not be blank and must start with slash" ?

File: server/sonar-web/src/main/webapp/WEB-INF/app/views/sessions/_form.html.erb (hunk 1 above)
Comment (julienlancelot) [inline server/sonar-web/src/main/webapp/WEB-INF/app/views/sessions/_form.html.erb]: Indeed !

File: sonar-plugin-api/src/main/java/org/sonar/api/server/authentication/Display.java (hunk 2 above)
Comment (julienlancelot) [inline sonar-plugin-api/src/main/java/org/sonar/api/server/authentication/Display.java]: I was asking myself the same question. So default value should be black ?

File: sonar-plugin-api/src/main/java/org/sonar/api/server/authentication/Display.java (hunk 3 above)
Comment (julienlancelot) [inline sonar-plugin-api/src/main/java/org/sonar/api/server/authentication/Display.java]: Ok

**Pull request merged**